### PR TITLE
Adding Cognitive Services Data Plane Vision+Language SDK

### PR DESCRIPTION
**Pull request by DavidLiCIG**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x ] Title of the pull request is clear and informative.
- [x ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@DavidLiCIG seems tests are failing on CI

File: src/SDKs/CognitiveServices/data-plane/Microsoft.CognitiveServices.Vision.Tests/Microsoft.CognitiveServices.Vision.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,43 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/data-plane/Microsoft.CognitiveServices.Vision.Tests/Microsoft.CognitiveServices.Vision.Tests.csproj]: @DavidLiCIG can you model your test project by looking at other tests.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language.Tests/BaseTests.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.CognitiveServices.Language.TextAnalytics;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using System;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.CognitiveServices.Language.Tests
```
Comment (arv100kri) [inline src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language.Tests/BaseTests.cs]: Shouldn't the namespace be Microsoft.CognitiveServices.Language.TextAnalytics.Tests? To better match the namespace of the actual API?

File: src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language.Tests/DetectLanguageTests.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                ITextAnalyticsAPI client = GetClient(HttpMockServer.CreateInstance());
+                LanguageBatchResultV2 result = client.DetectLanguage(
+                    new BatchInputV2(
+                        new List<InputV2>()
+                        {
+                        new InputV2("id","I love my team mates")
+                        }));
+
+                Assert.Equal("English", result.Documents[0].DetectedLanguages[0].Name);
+                Assert.Equal("en", result.Documents[0].DetectedLanguages[0].Iso6391Name);
```
Comment (arv100kri) [inline src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language.Tests/DetectLanguageTests.cs]: Also worth adding an assert for the score

**Review by arv100kri — APPROVED** [review]

File: src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language.Tests/BaseTests.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿using Microsoft.CognitiveServices.Language.TextAnalytics;
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language.Tests/BaseTests.cs]: @DavidLiCIG we have certain directory structure that is being used throughout the repo.
1) One being, if you have management and data plane the directories should be structured as
SDKs\CognitiveServices\management
SDKs\CognitiveServices\dataPlane
we rely on particular names to build and run tests during CI build.

Please change the directory structure accordingly.

File: src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    <DefineConstants>$(DefineConstants);CODESIGN</DefineConstants>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <AssemblyName>Microsoft.CognitiveServices.Language</AssemblyName>
+    <PackageTags>Microsoft Cognitive Services;Cognitive Services;Cognitive Services SDK;Text Analytics API;Text Analytics;REST HTTP client;netcore451511</PackageTags>
+    <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj]: @DavidLiCIG please remove GeneratePackageOnBuild, we have specific targets to build pacakges.
The target that we use is /t:CreateNugetPackage to create nuget packages, having it part of build system hampers the CI perf.

File: src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    <PackageReleaseNotes>For detailed release notes, see: </PackageReleaseNotes>
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj]: < packagereleasenotes > should be included on the top PropertyGroup where you have PackageId and version defined.
move it under right propertygroup.

File: src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(TargetFramework)'=='net452' ">
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/data-plane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj]: @DavidLiCIG remove line 20 to 32.
All these are already included for your projects.
Also we no longer use PORTABLE compiler constant in the entire repo. We have few traces of those constants being used in couple of RPs which are on track to be removed.

File: src/SDKs/CognitiveServices/data-plane/Vision/Microsoft.CognitiveServices.Vision.Tests/Microsoft.CognitiveServices.Vision.Tests.csproj [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    <None Update="SessionRecords\**\*.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Update="TestImages\detection1.jpg">
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/data-plane/Vision/Microsoft.CognitiveServices.Vision.Tests/Microsoft.CognitiveServices.Vision.Tests.csproj]: @DavidLiCIG combine items in Itemgroup starting line 27 to be part of ItemGroup starting line 17
Please organize items in limited item groups especially if they logically belong to same category

File: src/SDKs/CognitiveServices/data-plane/Vision/Microsoft.CognitiveServices.Vision/Microsoft.CognitiveServices.Vision.csproj [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
+    <PackageReleaseNotes>For detailed release notes, see: </PackageReleaseNotes>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(TargetFramework)'=='net452' ">
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/data-plane/Vision/Microsoft.CognitiveServices.Vision/Microsoft.CognitiveServices.Vision.csproj]: @DavidLiCIG please format project file according to the feedback given to the language project

File: src/SDKs/CognitiveServices/data-plane/dirs.proj [hunk 10]
```diff
@@ -0,0 +1,10 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/data-plane/dirs.proj]: @DavidLiCIG this file is not needed.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@DavidLiCIG travis changes have been pushed.
Please address the feedback and this PR is good to be merged

**Comment by DavidLiCIG** [comment]
@shahabhijeet Thanks. All feedbacks are addressed

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Forgot to mention earlier.
add < PackageReleaseNotes > and add Release notes.
I would recommend to use < ![CDATA and then format your Release notes

File: src/SDKs/CognitiveServices/dataPlane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj [hunk 11]
```diff
@@ -0,0 +1,24 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Microsoft.CognitiveServices.Language</PackageId>
+    <Description>This client library provides access to the Microsoft Cognitive Services Language APIs.</Description>
+    <VersionPrefix>1.0.0</VersionPrefix>
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj]: @DavidliCIG if this is a preview release, I still see your SDK nuget is not marked as preview
the version should 1.0.0-preview

File: src/SDKs/CognitiveServices/dataPlane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    <PackageTags>Microsoft Cognitive Services;Cognitive Services;Cognitive Services SDK;Text Analytics API;Text Analytics;REST HTTP client;netcore451511</PackageTags>
+    <PackageReleaseNotes>This is a preview release of the Cognitive Services Language SDK. Included with this release is support for Text Analytics API.</PackageReleaseNotes>
+    <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net452' ">
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj]: @DavidLigCIG please remove this redundant ItemGroup, this is not needed

File: src/SDKs/CognitiveServices/dataPlane/Vision/Microsoft.CognitiveServices.Vision/Microsoft.CognitiveServices.Vision.csproj [hunk 13]
```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Microsoft.CognitiveServices.Vision</PackageId>
+    <Description>This client library provides access to the Microsoft Cognitive Services Vision APIs.</Description>
+    <VersionPrefix>1.0.0</VersionPrefix>
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Vision/Microsoft.CognitiveServices.Vision/Microsoft.CognitiveServices.Vision.csproj]: @DavidLiCIG please mark this version as preview if this is a preview release. Release notes say this is preview but the version is not marked accordingly.
Developers who will be downloading from nuget will be confused what is the case, is it preview or not.

File: src/SDKs/CognitiveServices/dataPlane/Vision/Microsoft.CognitiveServices.Vision/Microsoft.CognitiveServices.Vision.csproj [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+    <PackageTags>Microsoft Cognitive Services;Cognitive Services;Cognitive Services SDK;Face API;REST HTTP client;Face SDK;netcore451511</PackageTags>
+    <PackageReleaseNotes>This is a preview release of the Cognitive Services Vision SDK. Included with this release is support for Face API.</PackageReleaseNotes>
+    <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net452' ">
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Vision/Microsoft.CognitiveServices.Vision/Microsoft.CognitiveServices.Vision.csproj]: Remove the Itemgroup as it's not needed.

File: src/SDKs/CognitiveServices/dataPlane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+  <PropertyGroup>
+    <PackageId>Microsoft.CognitiveServices.Language</PackageId>
+    <Description>This client library provides access to the Microsoft Cognitive Services Language APIs.</Description>
+    <VersionPrefix>1.0.0</VersionPrefix>
+    <DefineConstants>$(DefineConstants);CODESIGN</DefineConstants>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <AssemblyName>Microsoft.CognitiveServices.Language</AssemblyName>
+    <PackageTags>Microsoft Cognitive Services;Cognitive Services;Cognitive Services SDK;Text Analytics API;Text Analytics;REST HTTP client;netcore451511</PackageTags>
+    <PackageReleaseNotes>This is a preview release of the Cognitive Services Language SDK. Included with this release is support for Text Analytics API.</PackageReleaseNotes>
+    <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Language/Microsoft.CognitiveServices.Language/Microsoft.CognitiveServices.Language.csproj]: @DavidLiCIG generating documentation property is already applicable for the SDKs, this is again redundant, please remove

File: src/SDKs/CognitiveServices/dataPlane/Vision/Microsoft.CognitiveServices.Vision/Microsoft.CognitiveServices.Vision.csproj [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+  <PropertyGroup>
+    <PackageId>Microsoft.CognitiveServices.Vision</PackageId>
+    <Description>This client library provides access to the Microsoft Cognitive Services Vision APIs.</Description>
+    <VersionPrefix>1.0.0</VersionPrefix>
+    <DefineConstants>$(DefineConstants);CODESIGN</DefineConstants>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <AssemblyName>Microsoft.CognitiveServices.Vision</AssemblyName>
+    <PackageTags>Microsoft Cognitive Services;Cognitive Services;Cognitive Services SDK;Face API;REST HTTP client;Face SDK;netcore451511</PackageTags>
+    <PackageReleaseNotes>This is a preview release of the Cognitive Services Vision SDK. Included with this release is support for Face API.</PackageReleaseNotes>
+    <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Vision/Microsoft.CognitiveServices.Vision/Microsoft.CognitiveServices.Vision.csproj]: @DavidLiCIG please remove redundant property

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Did not see AssemblyInfo for individual SDKs which has AssemblyVersion and AssemblyFileVersion.
If you are missing you need those files.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**